### ISSUE TITLE
Disable plugin in '--production' mode

### ISSUE THREAD

**Issue by phtrivier** [brunch/brunch#1699]
Is there a way to disable a brunch plugin when running 'brunch build --production' ?

In my particular case (the 'brunch-flow' plugin), using 'overrides' does not work : 

```
  overrides: {
    production: {
      plugins: {
        flowtype: {
          enabled: false
        }
      }
    }
  }
```

I assume the 'enabled' option is plugin-specific, so unless the plugin implements it, this method should not work, right ?
Is there a more general way ?

**Comment by Florian-R** [comment]
You can use the `plugins.off` option conjointly with `overrides` : 

```js
overrides: {
  production: {
    plugins: {
      off: ['jade-brunch']
    }
  }
}
```

Some plugins also sometimes offers an ad-hoc option, but this is not idiomatic Brunch IMHO.

**Comment by shvaikalesh** [comment]
@Florian-R hey, thanks for the help!

> ad-hoc option, but this is not idiomatic Brunch IMHO.

could you please give an example?

**Comment by Florian-R** [comment]
TBH, I can't remember. I've seen this in a small plugin we used in my previous job, but I don't work there anymore so I can't check the sources.

**Comment by lydell** [comment]
auto-reload-brunch has an [`enabled` option](https://github.com/brunch/auto-reload-brunch/blob/98e1e835837603c24e9360a646a72a00f05fcec3/README.md#brunch-plugin-settings).

**Comment by shvaikalesh** [comment]
...and the `enabled` option is set by Brunch itself when in `production`, not the plugin. Will fix in #1664.
